sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -88,11 +88,13 @@ class V2SessionCatalog(catalog: SessionCatalog)
   override def loadTable(ident: Identifier): Table = {
     try {
       val table = catalog.getTableMetadata(ident.asTableIdentifier)
+      val qualifiedIdent = catalog.qualifyIdentifier(table.identifier)
```

Comment:
`catalog.getTableMetadata` should always return the `CatalogTable` with identifier qualified.